src/pseudopeople/loader.py

```diff
@@ -34,6 +34,10 @@ def load_and_prep_1040_data(data_path: dict, user_filters: List[Tuple]) -> pd.Da
 
     # Load data
     df_1040 = load_standard_dataset_file(data_path[DatasetNames.TAXES_1040], user_filters)
+    # We do not want to filter by state for dependents
+    for i in range(len(user_filters)):
+        if user_filters[i][0] == COLUMNS.mailing_state.name:
+            user_filters.pop(i)
```

Comment:
This is a bit dangerous b/c you're popping off the state from the user_filters permanently. I know currently it's fine b/c you're doing dependents last but I'd consider something more robust like making a copy of the list and working on that

Reply:
`user_filters = [f for f in user_filters if f[0] != COLUMNS.mailing_state.name]` can replace this loop, and make it non-destructive!

Sorry for being nosy, I get these emails and read them sometimes 😄

Reply:
In fact, I believe this loop has a bug -- if the state filter wasn't the last filter, when you pop it the list gets shorter, but you are still iterating to the same number. So line 39 fails with `IndexError: list index out of range`.

Reply:
Uff good call, @zmbc !

Reply:
Yes I was running the slow pytests that dont get run and they did catch this bug. But good catch.